lib/veri_traci.ml

```diff
@@ -0,0 +1,89 @@
+open Bap.Std
+module Bap_value = Value
+open Bap_traces.Std
+open Monad.State
+open Bil.Result
+
+type event = Trace.event
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+(** TODO: what is a behaviour of update_event method?  *)
+class context arch = object(self:'s)
+  inherit Bili.context
+  val mem  : var = mem_of_arch arch
```

Comment:
there is no need to introduce `mem` or `arch` as an instance variables. It looks like, that you're going to change them during the execution. Just let-bind them, e.g., 
```
class context arch =
   let mem = mem_of_arch in
   object 
      method arch = arch
      method mem = mem
   end
```